### PR TITLE
[fact] Create syntax module

### LINKED ISSUE #56: Make a syntax module
Right now we have this in _sanitize.py. We probably want to move this along with `_check_syntax` to a separate module.

This way we can better document the `repobee-sanitizer` syntax.

```python
START_BLOCK = "REPOBEE-SANITIZER-BLOCK"
REPLACE_WITH = "REPOBEE-SANITIZER-REPLACE-WITH"
END_BLOCK = "REPOBEE-SANITIZER-END"

SANITIZER_MARKERS = (
    START_BLOCK,
    REPLACE_WITH,
    END_BLOCK,
)
```

### PR DESCRIPTION
**Pull request by tohanss**
Fix #56

### CODE REVIEW

File: repobee_sanitizer/_syntaxhelpers.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+START_BLOCK = "REPOBEE-SANITIZER-START"
+REPLACE_WITH = "REPOBEE-SANITIZER-REPLACE-WITH"
+END_BLOCK = "REPOBEE-SANITIZER-END"
+
+SANITIZER_MARKERS = (
+    START_BLOCK,
+    REPLACE_WITH,
+    END_BLOCK,
+)
```
Comment (slarse) [inline repobee_sanitizer/_syntaxhelpers.py:23]: Let's put these into an enum.

```suggestion
import enum

class Marker(enum.Enum):
    """The markers that define sanitizer blocks."""
    START = "REPOBEE-SANITIZER-START"
    REPLACE = "REPOBEE-SANITIZER-REPLACE-WITH"
    END = "REPOBEE-SANITIZER-END"
```
To get the string from one of those, you do the following:

```python
print(Marker.START.value) # prints REPOBEE-SANITIZER-START
```
Note that you must access the string with `.value`. It's not enough to just do `Marker.START`.

Also, we can iterate over the markers like so:

```python
for marker in Marker:
    print(marker.value) # prints the marker strings
````

File: repobee_sanitizer/_syntaxhelpers.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
+"""Functions and constants relating to the syntax of repobee-sanitizer.
+
+.. module:: _syntaxhelpers
+    :synopsis: Functions and constants relating to the syntax of
+    repobee-sanitizer.
```
Comment (slarse) [inline repobee_sanitizer/_syntaxhelpers.py:5]: This line must be indented to count as part of the `synopsis` directive.
```suggestion
        repobee-sanitizer.
```

File: repobee_sanitizer/_syntax.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                (PREFIX REPLACE_MARKER
+                (PREFIX MARKERLESS_LINE)*)?
+                PREFIX END_MARKER
+            START_MARKER ::= "REPOBEE-SANITIZER-START\n"
+            REPLACE_MARKER ::= "REPOBEE-SANITIZER-REPLACE-WITH\n"
+            END_MARKER ::= "REPOBEE-SANITIZER-END\n"
+            MARKERLESS_LINE ::= line without sanitizer markers
+            PREFIX ::= a sequence of characters that is defined for each block
+                as any sequence that appears before the START_MARKER of that
+                particular PREFIXED_BLOCK.
```
Comment (slarse) [inline repobee_sanitizer/_syntax.py:54]: Two things here.

1. The contents of the block are too indented, 4 spaces should be removed from each line (i.e. dedent by 1 indentation). That's probably my fault, I didn't really check the indentation when I wrote the suggestion.
2. There must be a blank line after `.. code-block:: raw`, or `FILE ::=` will be interpreted as an option for the `code-block` directive.

Comment (tohanss) [reply repobee_sanitizer/_syntax.py:54]: 1. Thought about it but it looked wierd. However, by following 2 and adding a blank line, it now looks good

Comment (slarse) [reply repobee_sanitizer/_syntax.py]: Perfect.

**Pull request merged**